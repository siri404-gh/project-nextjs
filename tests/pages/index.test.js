import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { useRouter } from 'next/router';

import Index from '@/pages';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

const mockRouter = {
  push: jest.fn(),
};

useRouter.mockReturnValue(mockRouter);

jest.useFakeTimers();

describe('Index', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('section');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('Should render without props', () => {
    act(() => {
      render(<Index />, container);
    });
    expect(container).toMatchSnapshot();
  });

});
