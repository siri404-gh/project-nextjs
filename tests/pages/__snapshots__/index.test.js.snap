// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Should render without props 1`] = `
<section>
  <img
    alt="blog"
    class="img img--center"
    height="200"
    loading="lazy"
    src="/sections/dev.svg"
    title="blog"
    width="375"
  />
</section>
`;
